### ISSUE TITLE
Some problems about hyperparameters.

### ISSUE THREAD

**Issue by CCCCCaO** [chxy95/SRCNN#2]
Hi!
I find that when I use the `train.py`, the `loss = loss_fn(imgHR, label)`warned that the size of two tensors is mismatching.
The size of input tensor is [128, 1, 5, 5], and the target tensor is [128, 1, 21, 21]. The input tensor's dimension is wrong, so the MSE loss function can't work properly.
This is perhaps caused by some hyperparameters in `model.py`
I changed all the `stride` in `nn.Conv2d` to `1`, and maybe the `kernel_size` should be assigned to `f1`, `f2` or `f3` not the `stride` equaled to them.
This is the original version:
```python
class SRCNN_net(torch.nn.Module):
    def __init__(self, num_channels, f1=9, f2=1, f3=5):
        super(SRCNN_net, self).__init__()
        self.conv1 = nn.Conv2d(in_channels=num_channels, out_channels=64, 
                               kernel_size=f1, stride=1, padding=0, bias=True)
        self.conv2 = nn.Conv2d(in_channels=64, out_channels=32, 
                               kernel_size=1, stride=f2, padding=0, bias=True)
        self.conv3 = nn.Conv2d(in_channels=32, out_channels=1, 
                               kernel_size=5, stride=f3, padding=0, bias=True)
```
This is what i changed:
```python
class SRCNN_net(torch.nn.Module):
    def __init__(self, num_channels, f1=9, f2=1, f3=5):
        super(SRCNN_net, self).__init__()
        self.conv1 = nn.Conv2d(in_channels=num_channels, out_channels=64, 
                               kernel_size=f1, stride=1, padding=0, bias=True)
        self.conv2 = nn.Conv2d(in_channels=64, out_channels=32, 
                               kernel_size=f2, stride=1, padding=0, bias=True)
        self.conv3 = nn.Conv2d(in_channels=32, out_channels=1, 
                               kernel_size=f3, stride=1, padding=0, bias=True)
```
Then, all python scripts can work properly. so i thinking maybe sth wrong about the model, you can change it.

Regards,

**Comment by chxy95** [comment]
> Hi!
> I find that when I use the `train.py`, the `loss = loss_fn(imgHR, label)`warned that the size of two tensors is mismatching.
> The size of input tensor is [128, 1, 5, 5], and the target tensor is [128, 1, 21, 21]. The input tensor's dimension is wrong, so the MSE loss function can't work properly.
> This is perhaps caused by some hyperparameters in `model.py`
> I changed all the `stride` in `nn.Conv2d` to `1`, and maybe the `kernel_size` should be assigned to `f1`, `f2` or `f3` not the `stride` equaled to them.
> This is the original version:
> 
> ```python
> class SRCNN_net(torch.nn.Module):
>     def __init__(self, num_channels, f1=9, f2=1, f3=5):
>         super(SRCNN_net, self).__init__()
>         self.conv1 = nn.Conv2d(in_channels=num_channels, out_channels=64, 
>                                kernel_size=f1, stride=1, padding=0, bias=True)
>         self.conv2 = nn.Conv2d(in_channels=64, out_channels=32, 
>                                kernel_size=1, stride=f2, padding=0, bias=True)
>         self.conv3 = nn.Conv2d(in_channels=32, out_channels=1, 
>                                kernel_size=5, stride=f3, padding=0, bias=True)
> ```
> 
> This is what i changed:
> 
> ```python
> class SRCNN_net(torch.nn.Module):
>     def __init__(self, num_channels, f1=9, f2=1, f3=5):
>         super(SRCNN_net, self).__init__()
>         self.conv1 = nn.Conv2d(in_channels=num_channels, out_channels=64, 
>                                kernel_size=f1, stride=1, padding=0, bias=True)
>         self.conv2 = nn.Conv2d(in_channels=64, out_channels=32, 
>                                kernel_size=f2, stride=1, padding=0, bias=True)
>         self.conv3 = nn.Conv2d(in_channels=32, out_channels=1, 
>                                kernel_size=f3, stride=1, padding=0, bias=True)
> ```
> 
> Then, all python scripts can work properly. so i thinking maybe sth wrong about the model, you can change it.
> 
> Regards,

**Comment by chxy95** [comment]
Yes, you are right. 
I have changed the code.
Thanks.

**Comment by chxy95** [comment]
> Hi!
> I find that when I use the `train.py`, the `loss = loss_fn(imgHR, label)`warned that the size of two tensors is mismatching.
> The size of input tensor is [128, 1, 5, 5], and the target tensor is [128, 1, 21, 21]. The input tensor's dimension is wrong, so the MSE loss function can't work properly.
> This is perhaps caused by some hyperparameters in `model.py`
> I changed all the `stride` in `nn.Conv2d` to `1`, and maybe the `kernel_size` should be assigned to `f1`, `f2` or `f3` not the `stride` equaled to them.
> This is the original version:
> 
> ```python
> class SRCNN_net(torch.nn.Module):
>     def __init__(self, num_channels, f1=9, f2=1, f3=5):
>         super(SRCNN_net, self).__init__()
>         self.conv1 = nn.Conv2d(in_channels=num_channels, out_channels=64, 
>                                kernel_size=f1, stride=1, padding=0, bias=True)
>         self.conv2 = nn.Conv2d(in_channels=64, out_channels=32, 
>                                kernel_size=1, stride=f2, padding=0, bias=True)
>         self.conv3 = nn.Conv2d(in_channels=32, out_channels=1, 
>                                kernel_size=5, stride=f3, padding=0, bias=True)
> ```
> 
> This is what i changed:
> 
> ```python
> class SRCNN_net(torch.nn.Module):
>     def __init__(self, num_channels, f1=9, f2=1, f3=5):
>         super(SRCNN_net, self).__init__()
>         self.conv1 = nn.Conv2d(in_channels=num_channels, out_channels=64, 
>                                kernel_size=f1, stride=1, padding=0, bias=True)
>         self.conv2 = nn.Conv2d(in_channels=64, out_channels=32, 
>                                kernel_size=f2, stride=1, padding=0, bias=True)
>         self.conv3 = nn.Conv2d(in_channels=32, out_channels=1, 
>                                kernel_size=f3, stride=1, padding=0, bias=True)
> ```
> 
> Then, all python scripts can work properly. so i thinking maybe sth wrong about the model, you can change it.
> 
> Regards,

Yes, you are right.
I have changed the code.
Thanks.

**Comment by CCCCCaO** [comment]
Nice!
